explore/src/main/scala/explore/Explore.scala

```diff
@@ -5,22 +5,60 @@ package explore
 
 import scala.scalajs.js
 
+import cats.implicits._
+import crystal.react.implicits._
 import explore.Routing
 import explore.model.RootModel
 import japgolly.scalajs.react.extra.router._
 import japgolly.scalajs.react.vdom.VdomElement
 import japgolly.scalajs.react.vdom.html_<^._
 
 import js.annotation._
+import crystal.ViewF
+import cats.effect.IO
+import explore.model.RootModelRouting
+import explore.model.Page
 
 @JSExportTopLevel("Explore")
 object ExploreMain extends AppMain {
 
-  private val router = RouterWithProps(BaseUrl.fromWindowOrigin, Routing.config)
+  protected val (router, routerLogic) =
+    RouterWithProps.componentAndLogic(BaseUrl.fromWindowOrigin, Routing.config)
+
+  // This builds a RouterCtl that doesn't Broadcast when a new Page is set.
+  // Maybe we can send a PR to the RouterCtl to add a silent: Boolean parameter to set?
```

Comment:
I haven't looked 100% at how things are implemented in SeqExec with Diode. I think there the model is updated silently and rerenders are triggered by setting a new page in the router.

Here's it's the other way round: since the model state is just regular topmost React state, changes are propagated automatically. But we do a silent URL update to avoid doing it twice.

Reply:
I rolled back on this idea. It turns out we DO want the Router to trigger, otherwise it won't render according to the new path.

We now break the loop in `Routing`: we only modify the model if the inferred page from the model differs from the one that is trying to be set.

Reply:
👍